lib/jekyll-admin/server/theme.rb

```diff
@@ -0,0 +1,117 @@
+module JekyllAdmin
+  class Server < Sinatra::Base
+    namespace "/theme" do
+      get do
+        ensure_theme
+        json(
+          {
+            :name        => theme.name,
+            :version     => theme.version,
+            :authors     => theme.authors.join(", "),
+            :license     => theme.license,
+            :path        => at_root,
+            :directories => entries.map(&:to_api),
+          }
+        )
+      end
+
+      get "/*?/?:path.:ext" do
+        ensure_theme
+        json(
+          { :name => filename,
+            :path => file_path, }.merge!(contents).merge!(url_fields)
```

Comment:
merge base hash with additional hashes from respective methods and pass the resulting hash to be output as API points